### PR TITLE
Feature/Validate Ethereum address when adding contract to emissions dashboard

### PR DESCRIPTION
**Pull request by Stanley-Okwii**
## Description

This PR adds contract address validation when adding a contract to the dashboard. It implements tickets [#348](https://app.zenhub.com/workspaces/popcorn-603eb49d1a9a0b0011b49b36/issues/popcorndao/workspace/348) and #349 

## To Test
- Check out this branch
- Cd into the emission-dashboard directory and run `yarn dev`
- Run the contracts service
- Using an address of one of the ethereum addresses, add that address to the dashboard.

### CODE REVIEW

File: packages/ui/src/components/popcorn/emissions-dashboard/AddContractModal/index.tsx [hunk 1]
```diff
@@ -46,31 +53,34 @@ export const AddContractModal: React.FC<ContractModalProps> = ({ open, setOpen,
...
+                      <p className="text-sm text-gray-700 pb-1">
+                        Enter Ethereum Address of contract
+                      </p>
+                      <input
+                        type="text"
+                        onChange={(event) =>
+                          setContractAddress(event.target.value)
+                        }
+                        className="w-full pl-4 pr-10 py-2 text-sm leading-none border rounded-lg shadow-sm focus:outline-none focus:shadow-outline text-gray-500 font-light"
+                        placeholder="0xD634182479185918515"
```
Comment (mefuru) [inline packages/ui/src/components/popcorn/emissions-dashboard/AddContractModal/index.tsx:76]: Placeholder isn't a valid ethereum address - can we use this '0xa258c4606ca8206d8aa700ce2143d7db854d168c'

File: packages/emissions-dashboard/pages/index.tsx [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { NavBar } from '@popcorn/ui/components/popcorn/emissions-dashboard/NavBar/index';
 import { useRouter } from 'next/router';
 import React, { useEffect, useState } from 'react';
+import web3 from 'web3';
+import { useWeb3React } from '@web3-react/core';
+import { Web3Provider } from '@ethersproject/providers';
```
Comment (mefuru) [inline packages/emissions-dashboard/pages/index.tsx:6]: Seems redundant to add providers twice, here and also in `_app.tsx`

File: packages/ui/src/components/popcorn/emissions-dashboard/AddContractModal/index.tsx (hunk 1 above)
Comment (Stanley-Okwii) [reply packages/ui/src/components/popcorn/emissions-dashboard/AddContractModal/index.tsx:76]: > Placeholder isn't a valid ethereum address - can we use this '0xa258c4606ca8206d8aa700ce2143d7db854d168c'

Yeah, updated it. I had used one from the wireframes without validating it 😀. Thanks, nice catch...

File: packages/emissions-dashboard/pages/index.tsx (hunk 2 above)
Comment (Stanley-Okwii) [reply packages/emissions-dashboard/pages/index.tsx:6]: @mefuru , When working on this, I noticed that we probably need the providers in both places. There are subtle differences in how we use them.  Also in the index.tsx  file, is where I would say that we tap into the ethereum network.

When I tried removing the providers from _app.tsx, I got some errors on the UI. Plus, I noticed that the packages/app was using the same approach when interacting with the providers. 
https://github.com/popcorndao/workspace/blob/4db8c53c3ecfaad9a1eb858fc050fd41f99fcbe7/packages/app/pages/_app.tsx#L6-L7, https://github.com/popcorndao/workspace/blob/f15cc0bf377dcd3061a57bbad38d23394359ec92/packages/app/pages/staking/index.tsx#L1-L2 , https://github.com/popcorndao/workspace/blob/36f3567e3deb4d87f77b925f95d42c43378b4712/packages/app/pages/rewards-manager/index.tsx#L1-L4

Comment (mefuru) [reply packages/emissions-dashboard/pages/index.tsx:6]: Thanks for the clarification @Stanley-Okwii

**Review by mefuru — APPROVED** [review]

**Pull request merged**